src/app/pages/questions/components/question/text-input/text-input.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core'
+
+@Component({
+  selector: 'text-input',
+  templateUrl: 'text-input.component.html'
+})
+export class TextInputComponent implements OnInit {
+  @Output()
+  valueChange: EventEmitter<string> = new EventEmitter<string>()
+
+  @Input()
+  type
+
+  placeholder = 'Enter text'
+  isDateTime: string
+  dateFormat: string
+
+  ngOnInit() {
+    this.isDateTime = this.getIsDateTime()
+    if (this.isDateTime) this.dateFormat = this.getDateFormat()
+  }
+
+  getIsDateTime() {
+    return this.type.includes('date') || this.type.includes('time')
+  }
+
+  getDateFormat() {
+    const type = this.type.split('_')[0]
+    const format = this.type.split('_')[1]
+    switch (type) {
+      case 'datetime':
+        this.placeholder = 'Enter date and time'
+        if (format == 'seconds') return 'D MMM YYYY H:mm:s'
+        return 'D MMM YYYY H:mm'
```

Comment:
Would it better to use `DD` to give a leading 0?

Reply:
Made the changes.

Reply:
Could the `moment` library be used for this? This way of annotating time and dates may show some internationalisation issues. See [example use of moment in the app](
https://github.com/RADAR-base/RADAR-Questionnaire/blob/70360055dad9dd01aad69a6ea4e343d800d41114/src/app/pages/home/components/task-calendar/task-calendar.component.ts#L42)
and [a possible way of getting locale format](https://stackoverflow.com/a/53211160)

Reply:
Yes, I agree with @blootsvoets. I had the impression from https://github.com/RADAR-base/RADAR-Questionnaire/issues/176 that `moment` was already used so we can use it here as well?

Reply:
I didn't think about the internationalisation part. I'm using `ion-datetime` and this part specifies what date format to show for the date picker. It doesn't support internationalisation but there is a work around by specifying your own month names. I'll work on it.